src/treasure.js

```diff
@@ -0,0 +1,17 @@
+class Treasure {
+  constructor(col, row) {
+    this.x = col * 100;
+    this.y = row * 100;
+    this.width = SQUARE_SIDE;
+    this.height = SQUARE_SIDE;
+  }
+
+  draw() {
+    image(treasureImage, this.x, this.y, this.width, this.height);
+  }
+
+  setRandomPosition() {
+    this.x = Math.floor(Math.random() * (WIDTH - this.width));
```

Comment:
You could also have used the following logic to prevent the treasure from appearing over the grid lines!
`this.x = Math.floor(Math.random() * 10) * (WIDTH / 10);`